### DISCUSSION TITLE
InformationPanel portaled to the root like Modal

### DISCUSSION THREAD

**Discussion by Juliakas** [iTwin/iTwinUI#1812] [Ideas]
The idea is to allow additional options for `InformationPanel` - instead of being displayed next to your content it could be portaled into application root - when opened it would expand from the side of the browser window - similar to how `Modal` is being portaled into application root (child of `ThemeProvider` by default) and displays over everything.

While this can be implemented by portaling InformationPanel ourselves, it can become cumbersome, especially when having a package that may want to use these root information panels - I would need to pass down my portal ref and write same logic inside the package - or at least create a common package. Plus having a standard behavior is always a plus :)

I would rather not post our use cases directly here, reach out to me if needed 😄

**Comment by mayank99** [comment]
My suggestion right now would be to use the built-in `<dialog>` element. It will automatically show in the top-layer (avoiding the need for portaling) and also automatically handle some built-in accessibility considerations (such as moving focus in/out of the dialog).

I created an example to show how you can use `<InformationPanel as="dialog">`: https://stackblitz.com/edit/github-oxw1gt?file=src%2FApp.tsx

**Reply by Juliakas** [reply]
Thanks for suggestion, I will try it out. We need to be able to click outside of panel which would be restricted with `<dialog>` though, if we can't override this behavior, we will probably stick with our implementation. If so, any considerations in the future to implement some functionality so that we could effectively have a side panel component?
